### DISCUSSION TITLE
Kamp not working/trouble shooting?

### DISCUSSION THREAD

**Discussion by kingmonkeyno1** [LeeOtts/Ender3v2-Klipper-Configs#18] [Q&A] [answered]
Hi there,
Ive attempted to run this config on my ender 3. Im running the 2.2.7 board and sprite extruder. Cant seem to get the adaptive mesh working in kamp. I think ive followed all the steps but it doesnt seem to do the adaptive mesh before a print.

I note that there is a link to the kamp repository, and on there , there is a kamp settings cfg. Do i need to download this or is everything included in this config here?

**Comment by LeeOtts** [comment, marked as answer]
You shouldn't need any other files. 

It sounds like you missed step 3 in the getting started on the readme. I had this happen when upgrading from a pi 3 to pi 4 and I forgot to make that change. 

Ensure the following is defined in moonraker.conf: [file_manager] enable_object_processing: True

**Reply by kingmonkeyno1** [reply]
Thanks for taking the time to reply.

Yeah i double checked all those things.

I tried everything at the time but no luck. I just gave up on it whilst i worked on some other issues. I even tried adding the M117 command as suggested elsewhere.

Well last evening (admittedly after making a new printer profile in cura) and adding the M117 command as the first line, it came back to life and began the kamp process, it was quite a nice surprise!  Maybe it was something to do with my printer settings .

Thanks for providing this config and contribution to the community!

**Reply by LeeOtts** [reply]
You are welcome. I'm happy to help. I love my Ender 3 and want others to enjoy theirs as much. 

Hmmm... Interesting. I haven't heard of this issue just yet. 

You mentioned adding a m117 command. If someone also has this error, where did you put the m117 order? Starting gcode?

Did you add the starting gcode I documented first?

**Reply by kingmonkeyno1** [reply]
Im relatively new to 3d printing, and trying to learn all the basics on the ender 3 and maximising its potential. I know there are more easier options out there these days!

So, just to clarify, that might have been the issue thinking about it. In my earlier failed attempts i failed to use your start gcode.

In the new profile , i did. So perhaps the M117 was not even relevant .

I used the M117 code from the KAMP trouble shooting guide;

"No matter what, meshes and purges are not adapting

This happens when Moonraker's object processor is putting exclude object definitions in the wrong spot (AFTER Print_Start rather than before). This is usually fixed if you add M117 before your print_start macro in your slicer's starting gcode. M117 is just a basic "clear display" gcode, but will force the preprocessor to place the definition macros in the correct spot."

I just put M117 as the first line. Not sure if this is correct btw.

Im using cura , for context.

**Reply by LeeOtts** [reply]
No worries we all gotta start somewhere. I started on a Ender 3 now I work in 3d printing as a career for a large manufacturer. 

It could have been the start gcode but did you also try to install kamp? You mentioned the KAMP settings file. If you used the kamp adative meshing files on the kamp github it will not work with this config. They have continued updating since I made this cofig while mine has pretty much stayed the same. It has to be used as intended.
